ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/EditorConfigInternal.kt

```diff
@@ -8,8 +8,8 @@ import java.util.Properties
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.ConcurrentHashMap
 
-class EditorConfig private constructor (
-    val parent: EditorConfig?,
+class EditorConfigInternal private constructor (
```

Comment:
Totally open for better names for this class.

Reply:
What's wrong with just EditorConfig ;) ?

Reply:
@arturbosch there's already an `EditorConfig` interface inside of `ktlint-core` so if you want to use one or the other you need a lot of ugly namespacing. :) We can also rename that one, but I couldn't think of a name for that one!

Reply:
Oh, didn't know that.
I like to use a `Default`for my implementations of an interface.
So maybe consider `DefaultEditorConfig`.

Reply:
@arturbosch but that's the thing - the internal one is actually not an implementation of the `EditorConfig` interface ;)

Reply:
haha correct! I don't have context about why it was implemented this way. I thought about naming the internal one `ParsedEditorConfig` or `EditorConfigParser` or something...